src/components/dialogs/MutedWords.tsx

```diff
@@ -0,0 +1,317 @@
+      setContentTargetEnabled(false)
+      setTagTargetEnabled(false)
+    } catch (e: any) {
+      logger.error(`Failed to save muted word`, {message: e.message})
+      setError(e.message)
+    }
+  }, [field, tagTargetEnabled, contentTargetEnabled, addMutedWord, setField])
+
+  return (
+    <Dialog.ScrollableInner label={_(msg`Manage your muted words and tags`)}>
+      <Text
+        style={[a.text_md, a.font_bold, a.pb_sm, t.atoms.text_contrast_high]}>
+        <Trans>Add muted words and tags</Trans>
+      </Text>
+      <Text style={[a.pb_lg, a.leading_snug, t.atoms.text_contrast_medium]}>
+        <Trans>
+          Posts can be muted based on their text, their tags, or both.
+        </Trans>
+      </Text>
+
+      <View style={[a.pb_lg]}>
+        <Dialog.Input
+          autoCorrect={false}
+          autoCapitalize="none"
+          autoComplete="off"
+          label={_(msg`Enter a word or tag`)}
+          placeholder={_(msg`Enter a word or tag`)}
+          value={field}
+          onChangeText={setField}
+          onSubmitEditing={submit}
+        />
+
+        <View
+          style={[
+            a.pt_sm,
+            a.mb_lg,
+            a.flex_row,
+            a.align_center,
+            a.gap_md,
+            a.flex_wrap,
+          ]}>
+          <Toggle.Item
+            label={_(msg`Mute this word in post text`)}
+            name="content"
+            value={contentTargetEnabled}
+            onChange={setContentTargetEnabled}
+            style={[a.flex_1]}>
+            <TargetToggle>
+              <View style={[a.flex_row, a.align_center, a.gap_sm]}>
+                <Toggle.Checkbox />
+                <Toggle.Label>
+                  <Trans>Mute in text</Trans>
+                </Toggle.Label>
+              </View>
+              <PageText size="sm" />
+            </TargetToggle>
+          </Toggle.Item>
+
+          <Toggle.Item
+            label={_(msg`Mute this tag`)}
+            name="tag"
+            value={tagTargetEnabled}
+            onChange={setTagTargetEnabled}
+            style={[a.flex_1]}>
+            <TargetToggle>
+              <View style={[a.flex_row, a.align_center, a.gap_sm]}>
+                <Toggle.Checkbox />
+                <Toggle.Label>
+                  <Trans>Mute in tags</Trans>
+                </Toggle.Label>
+              </View>
+              <Hashtag size="sm" />
+            </TargetToggle>
+          </Toggle.Item>
+
+          <Button
+            disabled={
+              !field ||
+              Boolean(field && !contentTargetEnabled && !tagTargetEnabled)
+            }
+            label={_(msg`Add mute word for configured settings`)}
+            size="small"
+            color="primary"
+            variant="solid"
+            style={[!gtMobile && [a.w_full, a.flex_0]]}
+            onPress={submit}>
+            <ButtonText>
+              <Trans>Add</Trans>
+            </ButtonText>
+            <ButtonIcon icon={Plus} />
+          </Button>
```

Comment:
Should probably disable this and display a spinner or something in the button while we save the new tag.